File: 자프링-GraphQL-실습/src/main/java/com/junyss/graphqltest/api/supply/resolver/SupplyResolver.java
package com.junyss.graphqltest.api.supply.resolver;

import java.util.List;

import org.springframework.graphql.data.method.annotation.Argument;
import org.springframework.graphql.data.method.annotation.MutationMapping;
import org.springframework.graphql.data.method.annotation.QueryMapping;
import org.springframework.stereotype.Controller;

import com.junyss.graphqltest.api.supply.model.dto.request.SupplyRequestDto;
import com.junyss.graphqltest.api.supply.model.dto.response.SupplyResponseDto;
import com.junyss.graphqltest.api.supply.service.SupplyService;
import com.junyss.graphqltest.common.constant.DefaultResponse;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Controller
public class SupplyResolver {

	private final SupplyService supplyService;

	@MutationMapping
	public DefaultResponse<String> saveForSupply (@Argument("input") SupplyRequestDto supplyRequestDto) {
		return supplyService.saveForSupply(supplyRequestDto);
	}

	@QueryMapping
	public DefaultResponse<List<SupplyResponseDto>> getSupplyList (
		@Argument Integer page,
		@Argument Integer size) {

		return supplyService.getSupplyList(page, size);
	}

	@QueryMapping
	public DefaultResponse<SupplyResponseDto> getSupplyBySupplyId (@Argument String supplyId) {
		return supplyService.getSupplyBySupplyId(supplyId);
	}

	@QueryMapping
	public DefaultResponse<List<SupplyResponseDto>> getSupplyByTeamId (
		@Argument Long teamId,
		@Argument Integer page,
		@Argument Integer size) {

		return supplyService.getSupplyByTeamId(teamId, page, size);
	}

	@MutationMapping
	public DefaultResponse<String> deleteSupply (@Argument String supplyId) {
		return supplyService.deleteSupply(supplyId);
	}
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/supply/resolver/SupplyResolver.kt
package com.junyharangstudy.kotlingraphqltest.api.supply.resolver

import com.junyharangstudy.kotlingraphqltest.api.common.constant.PageRequestDto
import com.junyharangstudy.kotlingraphqltest.api.software.model.dto.response.SoftwareResponseDto
import com.junyharangstudy.kotlingraphqltest.api.supply.model.dto.request.SupplyCreateRequestDto
import com.junyharangstudy.kotlingraphqltest.api.supply.model.dto.response.SupplyResponseDto
import com.junyharangstudy.kotlingraphqltest.api.supply.service.SupplyService
import com.junyharangstudy.kotlingraphqltest.common.constant.DefaultResponse
import org.springframework.graphql.data.method.annotation.Argument
import org.springframework.graphql.data.method.annotation.MutationMapping
import org.springframework.graphql.data.method.annotation.QueryMapping
import org.springframework.stereotype.Controller

@Controller
class SupplyResolver (private val supplyService: SupplyService) {
    @MutationMapping
    fun saveSupply(@Argument("input") supplyCreateRequestDto: SupplyCreateRequestDto): DefaultResponse<String> {
        return supplyService.saveSupply(supplyCreateRequestDto)
    }

    @QueryMapping
    fun getSupplyList(
        @Argument teamId: Long?,
        @Argument pageRequestDto: PageRequestDto?
    ): DefaultResponse<List<SupplyResponseDto>> {
        return supplyService.getSupplyList(teamId, pageRequestDto)
    }

    @QueryMapping
    fun getSupplyBySupplyId (@Argument supplyId: String): DefaultResponse<SupplyResponseDto> {
        return supplyService.getSupplyBySupplyId(supplyId)
    }

    @MutationMapping
    fun updateSupply(
        @Argument("supplyId") supplyId: String,
        @Argument("teamId") teamId: Long
    ): DefaultResponse<String> {
        return supplyService.updateSupply(supplyId, teamId)
    }

    @MutationMapping
    fun deleteSupply(@Argument supplyId: String): DefaultResponse<String> {
        return supplyService.deleteSupply(supplyId)
    }
}